src/IB/IBFEInstrumentPanel.cpp

```diff
@@ -330,9 +351,85 @@ IBFEInstrumentPanel::readInstrumentData(const int U_data_idx,
+            const Box<NDIM> domain_box_level = Box<NDIM>::refine(domain_box, ratio);
+            const hier::Index<NDIM>& domain_box_level_lower = domain_box_level.lower();
+            const hier::Index<NDIM>& domain_box_level_upper = domain_box_level.upper();
+            std::array<double, NDIM> dx;
+            for (unsigned int d = 0; d < NDIM; ++d)
+            {
+                dx[d] = dx_coarsest[d] / static_cast<double>(ratio(d));
+            }
+            const auto i = IndexUtilities::getCellIndex(&x_centroid(0),
+                                                        domain_x_lower,
+                                                        domain_x_upper,
+                                                        dx.data(),
+                                                        domain_box_level_lower,
+                                                        domain_box_level_upper);
+            Box<NDIM> cell_box(i, i);
+            tbox::Array<int> local_patch_idxs;
+            box_tree->findLocalOverlapIndices(local_patch_idxs, cell_box);
+            if (local_patch_idxs.size() != 0)
+            {
+                TBOX_ASSERT(local_patch_idxs.size() == 1);
+                x_centroid_local_patch_idx = local_patch_idxs[0];
+                x_centroid_local_ln = ln;
+            }
+        }
+        x_centroid_global_ln = IBTK_MPI::maxReduction(x_centroid_local_ln);
+        if (x_centroid_global_ln == x_centroid_local_ln)
+        {
+            Pointer<PatchLevel<NDIM> > level = hierarchy->getPatchLevel(x_centroid_local_ln);
+            if (x_centroid_local_patch_idx != IBTK::invalid_index)
+            {
+                Pointer<Patch<NDIM> > patch = level->getPatch(x_centroid_local_patch_idx);
+                Pointer<CellData<NDIM, double> > p_cc_data = patch->getPatchData(P_data_idx);
+                if (p_cc_data)
+                {
+                    LEInteractor::interpolate(&d_centroid_pressure_values[meter_idx],
+                                              1,
+                                              1,
+                                              &x_centroid(0),
+                                              NDIM,
+                                              NDIM,
+                                              p_cc_data,
+                                              patch,
+                                              p_cc_data->getGhostBox(),
+                                              d_p_interp_fcn);
+                }
+                else
+                {
+                    TBOX_ERROR("no pressure data!\n");
+                }
+            }
+        }
+    }
+
     // Synchronize the values across all processes.
     IBTK_MPI::sumReduction(d_flow_rate_values.data(), d_num_meters);
     IBTK_MPI::sumReduction(d_mean_pressure_values.data(), d_num_meters);
+    IBTK_MPI::sumReduction(d_centroid_pressure_values.data(), d_num_meters);
```

Comment:
While this is correct I would still prefer to broadcast unique values instead of doing a reduction. Perhaps we can improve this later.

Reply:
NP. I just made this change.